core/src/graph/config-graph.ts

```diff
@@ -427,6 +427,7 @@ export abstract class BaseConfigGraph<
   }
 
   protected addActionInternal(action: Action) {
+    this.actions[action.kind][action.name] = <any>action
```

Comment:
The existing conditional type can be applied instead of casting to`<any>`:
```suggestion
  protected addActionInternal<K extends ActionKind>(action: Action) {
    this.actions[action.kind][action.name] = <PickTypeByKind<K, B, D, R, T>>action
```